WordPress/src/main/java/org/wordpress/android/util/encryption/EncryptionUtils.kt

```
@@ -0,0 +1,88 @@
+package org.wordpress.android.util.encryption
+
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import org.wordpress.android.modules.BG_THREAD
+import java.security.Key
+import java.security.KeyStore
+import java.util.Base64
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+import javax.crypto.spec.GCMParameterSpec
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+
+private const val KEY_STORE_ALIAS = "AndroidJPSecretKey"
+private const val PROVIDER_NAME = "AndroidKeyStore"
+private const val CIPHER_TRANSFORMATION_TYPE = "AES/GCM/NoPadding"
+private const val T_LENGTH = 128
+
+@Singleton
+class EncryptionUtils @Inject constructor(
+    @param:Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
+) {
+    private val secretKey: Key by lazy {
+        getKeyFromStore() ?: initSecretKey()
+    }
+
+    @Suppress("TooGenericExceptionCaught", "SwallowedException")
+    private fun getKeyFromStore(): Key? {
+        return try {
+            val ks: KeyStore = KeyStore.getInstance(PROVIDER_NAME).apply {
+                load(null)
+            }
+            ks.getKey(KEY_STORE_ALIAS, "".toCharArray())
+        } catch (ignore: Throwable) {
+            null
+        }
```

What is the expected behavior when we get an exception here?

It looks to me like we'll generate a new key after each exception and if I am not mistaken, that means the previously encrypted data will not be accessible to us. Assuming that's correct, I think this can lead to a frustrating experience for some of the users because we won't be aware of it unless the user reports it to us as a support request.

We have a few options:
1. Keep the current behavior, but log the exception, so it'll be visible during support requests, crash reports etc
2. Keep the current behavior, but [send a report to Sentry](https://github.com/Automattic/Automattic-Tracks-Android/blob/trunk/crashlogging/src/main/java/com/automattic/android/tracks/crashlogging/CrashLogging.kt#L36-L52)
3. Change the behavior to crash on unexpected exceptions

In general, I prefer the fail fast approach in `3`, but I am not sure if it's the right choice for this flow or not, because I am not sure what our planned final flow looks like. I'll chat with @jkmassel to get a better handle on that.

---

Actually, having an exception means that the Key is not present in the system, so we are then, returning null and forcing the generation of a new one. However, your point is right when saying we can be hiding other exceptions.

Let me see if we can detect when the key is not present, so we are not returning an exception in that case, but crashing the app in other scenarios

---

Ok, I misread the docs, so in case the key is not present in the system, no exception is thrown, and we are getting just null.
So, I've removed the try-catch block altogether, and we should be good to go now with exception reports.

PS: I've also removed all the testing code since you have already tested the PR